train_gpt2.cu

```diff
@@ -325,6 +325,7 @@ typedef struct {
     int* targets; // the target tokens for the current forward pass
     float mean_loss; // after a forward pass with targets, will be populated with the mean loss
     float accumulated_mean_loss; // Mean loss after aggregating it on all GPUs
+    float* unified_buffer; // GPU buffer to avg loss across process
```

Comment:
doesn't seem like unified buffer should be part of model, possible for it to maybe live "outside" inside int main or as global inside zero.cuh file or so?